frontend/lib/src/components/shared/BaseButton/styled-components.ts

```diff
@@ -332,6 +337,45 @@ export const StyledPillsButtonActive = styled(
   }
 })
 
+export const StyledSegmentedControlButton = styled(
+  StyledButtonGroupBaseButton
+)<RequiredBaseButtonProps>(({ theme }) => {
+  return {
+    padding: `${theme.spacing.twoXS} ${theme.spacing.lg}`,
+    borderRadius: theme.radii.none,
+    flex: "1 0 fit-content",
+    maxWidth: "100%",
+    marginRight: `-${theme.sizes.borderWidth}`, // Add negative margin to overlap borders
+
+    "&:first-child": {
+      borderTopLeftRadius: theme.radii.default,
+      borderBottomLeftRadius: theme.radii.default,
+    },
+    "&:last-child": {
+      borderTopRightRadius: theme.radii.default,
+      borderBottomRightRadius: theme.radii.default,
+      marginRight: theme.spacing.none, // Reset margin for the last child
+    },
+    "&:hover": {
+      zIndex: 100, // Make sure overlapped borders are visible
+    },
+  }
+})
+
+export const StyledSegmentedControlButtonActive = styled(
+  StyledSegmentedControlButton
+)<RequiredBaseButtonProps>(({ theme }) => {
+  return {
+    backgroundColor: transparentize(theme.colors.primary, 0.9),
+    borderColor: theme.colors.primary,
+    color: theme.colors.primary,
+    zIndex: 100,
```

Comment:
this value should come from the theme.zIndices object. It would be best to use an existing zIndex, but if there is no suitable one, feel free to add a new. In context of the Advanced Theming project, we try to get rid of as many hard-coded values as possible 🙂

Reply:
maybe we can just use zIndices.priority here?